### ISSUE TITLE
Future of this package after PHP 8.1

### ISSUE THREAD

**Issue by Gummibeer** [spatie/enum#90]
Now that the native PHP enums RFC is accepted we have to think about the future of this package.

https://wiki.php.net/rfc/enumerations#voting

Following the RFC I think that there's nothing this package provides on top except of labels.
And some static methods like the `toArray()` `toValues()` ...

The Laravel wrapper package has some more features we should keep. So a custom interface including abstract class with our on-top logic could be nice.

Question is if we want this as a dedicated base package or move it into the Laravel wrapper?

**Comment by brendt** [comment]
I think it's best to archive this package, and move the little functionality that's left to the Laravel package. It's been a good run :)

**Comment by Gummibeer** [comment]
While digging deeper I think that there are still some features that several users need/want to use and would be worth keeping this package up - labels and to array methods.

I would like to wait for first RC with native enums and update the packages code with it.
After this we will see what stays in the package and if it's worth it.

**Comment by brendt** [comment]
We definitely don't need to rush sunsetting this package. From my point of view I'll switch to native enums though. Regarding labels I'd write them like so:

```php
enum Status
{
	case draft;
	case published;
	case archived;

	public function label(): string
	{
		return match($this)
		{
			self::draft => 'Draft',
			self::published => 'Published',
			self::archived => 'Archived',
		};
	}
}
```

`toArray` is also covered with the built-in `Status::cases()` method. I'd probably writing something like this:

```php
$labels = array_map(
	fn(Status $status) => $status->label(),
	Status::cases(),
);
```

For _my_ use cases, I'm happy to work around the missing functionality with a little trait, but I don't think I'd need a package anymore. I'm fine to keep supporting it though if there's really a need for it.

**Comment by Gummibeer** [comment]
I think that traits and interfaces would be a good solution. We will need the interface for the assertions and the faker provider.
Primary these two classes are gold - I use them a lot in my apps, as they make writing tests with enums a lot easier.

**Comment by brendt** [comment]
I agree, the interface + trait combo might add significant value. Let's revisit this when 8.1 beta or RC builds are available :)
